pkg/controller.v1/pytorch/hpa.go

```diff
@@ -45,21 +47,25 @@ func (r *PyTorchJobReconciler) ReconcileHPA(pytorchJob *kubeflowv1.PyTorchJob) e
 		return err
 	}
 
-	if err := r.Get(context.TODO(), types.NamespacedName{
-		Name:      pytorchJob.Name,
-		Namespace: pytorchJob.Namespace,
-	}, current); err != nil {
-		if !errors.IsNotFound(err) {
-			return err
+	err = r.Get(context.TODO(), client.ObjectKeyFromObject(expected), current)
+	if err != nil {
+		if errors.IsNotFound(err) && !trainutil.IsJobSuspended(&pytorchJob.Spec.RunPolicy) {
+			// Create the new HPA.
+			logger.V(1).Info("Creating HPA", "namespace", expected.Namespace, "name", expected.Name)
+			err = r.Create(context.TODO(), expected)
+			if err != nil {
+				return err
+			}
+			return nil
 		}
-
-		// Create the new HPA.
-		logger.V(1).Info("Creating HPA", "namespace", expected.Namespace, "name", expected.Name)
-		err = r.Create(context.TODO(), expected)
-		if err != nil {
+		return err
+	}
+	if trainutil.IsJobSuspended(&pytorchJob.Spec.RunPolicy) {
+		// Delete the current HPA
+		logger.V(1).Info("Deleting HPA", "HorizontalPodAutoscaler", klog.KObj(current))
+		if err = r.Delete(context.TODO(), current); err != nil {
```

Comment:
```suggestion
		return r.Delete(context.TODO(), current); err != nil {
```
IIUC if the delete succeeds there is not need for the further reconciliation - the update code below could result in an unnecessary error.

Reply:
That's correct. Thanks.

Reply:
Done.